src/axom/quest/examples/quest_signed_distance_interface.cpp

```diff
@@ -52,21 +55,74 @@ int numranks;
 /*!
  * \brief Holds command-line arguments
  */
-static struct
+struct Arguments
 {
   std::string fileName;
-  int ndims;
-  int maxLevels;
-  int maxOccupancy;
-  int box_dims[3];
-  double box_min[3];
-  double box_max[3];
-  bool specified_box_min;
-  bool specified_box_max;
-  bool is_water_tight;
-  bool dump_vtk;
-  bool use_shared;
-} Arguments;
+  int ndims{3};
+  int maxLevels{15};
+  int maxOccupancy{5};
+  std::vector<int> box_dims{32,32,32};
+  std::vector<double> box_min;
+  std::vector<double> box_max;
+  bool is_water_tight{true};
+  bool dump_vtk{true};
+  bool use_shared{false};
+  bool use_batched_query{false};
+  bool ignore_signs{false};
+
+  void parse( int argc, char** argv, CLI::App& app)
+  {
+    app.add_option("-f,--file", this->fileName, "specifies the input mesh file")
+      ->check(CLI::ExistingFile)
+      ->required();
+
+    app.add_option("--dimension", this->ndims, "the problem dimension")
+      ->capture_default_str();
+
+    app.add_option("--maxLevels", this->maxLevels, "max levels of Subdivision for the BVH")
+      ->capture_default_str();
+
+    app.add_option("--maxOccupancy", this->maxOccupancy, "max number of item per BVH bin")
+      ->capture_default_str();
+
+    app.add_option("--box-dims", box_dims, "the dimensions of the box mesh")
+      ->expected(3);
+
+    auto* minbb = app.add_option("--box-min", box_min, "the lower corner of the box mesh")
+      ->expected(3);
+    auto* maxbb = app.add_option("--box-max", box_max, "the upper corner of the box mesh")
+      ->expected(3);
+    minbb->needs(maxbb);
+    maxbb->needs(minbb);
+
+    app.add_flag("!--no-vtk", this->dump_vtk, "disables VTK output")
+      ->capture_default_str();
+
+    app.add_flag("!--not-watertight", this->is_water_tight, "indicates that input is not water-tight")
+      ->capture_default_str();
+
+    app.add_flag("--use-shared", this->use_shared, "stores the surface using MPI-3 shared memory")
+      ->capture_default_str();
+
+    app.add_flag("--batched", this->use_batched_query, 
+      "uses a vectorized query instead of many serial queries")
```

Comment:
I wouldn't call this `vectorized` -- just call it `parallel` or `parallel via OpenMP` to be more precise..

Reply:
Technically, it is actually not necessarily `parallel` or `parallel via OpenMP` -- that depends on how axom is configured.
The only guarantee is that it is batched.

I'll try to rework this message.